src/QuadForm/Herm/Lattices.jl

```diff
@@ -321,14 +320,15 @@ end
 ################################################################################
 
 function dual(L::HermLat)
-  G, B = _gram_schmidt(gram_matrix_of_rational_span(L), involution(L))
+  G = gram_matrix_of_rational_span(L)
+  @req rank(G) == nrows(G) "Lattice must be non-degenerate"
```

Comment:
```
 rank(L::AbsLat) -> Int

  Returns the rank of the underlying module of L.
```
Hence this is not the correct test. I propose introducing a (cached) method `isdegenerate(L::AbsLat)` which returns `rank(L) == rank(rational_span(L))`

Reply:
I am testing whether the Gram matrix has full rank. Why is this not correct?

Reply:
ah I missred the lhs for `rank(L)` so nvm it is correct